etl/src/main/java/org/pih/hivmigration/etl/sql/AccompagnateurMigrator.groovy

```diff
@@ -0,0 +1,39 @@
+package org.pih.hivmigration.etl.sql
+
+class AccompagnateurMigrator extends ObsMigrator {
+
+    @Override
+    def void migrate() {
+
+        create_tmp_obs_table()
+
+        executeMysql("Load Accompagnateur names as observations on the most recent followup or intake form", ''' 
+                
+        INSERT INTO tmp_obs (
+            value_text, 
+            source_patient_id, 
+            source_encounter_id, 
+            concept_uuid)                
+        SELECT 
+            p.accompagnateur_name, 
+            e.source_patient_id, 
+            e.source_encounter_id, 
+            concept_uuid_from_mapping('CIEL', '164141') as concept_uuid
+        FROM hivmigration_patients p 
+        INNER JOIN hivmigration_encounters e 
+            on p.source_patient_id = e.source_patient_id and p.accompagnateur_name is not null
+        INNER JOIN
+            (SELECT source_patient_id, MAX(encounter_date) AS MaxEncounterDate
+            FROM hivmigration_encounters where  source_encounter_type in ('intake', 'followup') 
+            GROUP BY source_patient_id) ee 
+        ON e.source_patient_id = ee.source_patient_id 
+        AND e.encounter_date = ee.MaxEncounterDate 
+        and e.source_encounter_type in ('intake', 'followup');
```

Comment:
Why not just

```suggestion
            (SELECT source_encounter_id, MAX(encounter_date) AS MaxEncounterDate
            FROM hivmigration_encounters where  source_encounter_type in ('intake', 'followup') 
            GROUP BY source_patient_id) ee 
        ON e.source_encounter_id = ee.source_encounter_id;
```

?

Reply:
Whoops, because that wouldn't work, is why :laughing:

Reply:
Thanks @brandones for reviewing this. I initially had the same inner query on the source_encounter_id you suggested, but I got incorrect results. It returned a mismatch between the max(encounter_date) and the source_encounter_id. I spent couple hours debugging these SQL queries.